src/ui/Avatar.cpp

```diff
@@ -1,17 +1,28 @@
 #include <QPainter>
 #include <QPainterPath>
-#include <QSettings>
+#include <QSvgRenderer>
 
 #include "AvatarProvider.h"
+#include "UserSettingsPage.h"
 #include "Utils.h"
+#include "jdenticoninterface.h"
 #include "ui/Avatar.h"
 
 Avatar::Avatar(QWidget *parent, int size)
   : QWidget(parent)
   , size_(size)
+  , jdenticonInterface_{Jdenticon::getJdenticonInterface()}
 {
-        type_   = ui::AvatarType::Letter;
-        letter_ = "A";
+        if (UserSettings::instance()->useIdenticon())
```

Comment:
Should check that the plugin is available in addition to checking the setting

Reply:
Hmm, I think I fixed this somewhere else already. Thanks for pointing that out!

Reply:
Actually, the reason I did this is because the useIdenticon() function only returns true if both the user has turned identicons on and if the plugin is available.